### PR TITLE
resolve PATHEXT on Windows

### LINKED ISSUE #640: The executable python does not exist
Running `virtualenv myvenv --python=python` fails with "The executable python (from --python=python) does not exist" even if python is on the PATH:

```
$ which python
/c/Python27/python
```

The issue is wrong executable detection on Windows. `python` is equivalent to `python.exe` on Windows but this is not recognized by virtualenv. It works only if you use `virtualenv myvenv --python=python.exe`.

### PR DESCRIPTION
**Pull request by schlamar**
This fixes #640 for me.

It works now with `virtualenv --python=C:\Python27\python` and `virtualenv --python=python` on Windows.

### CODE REVIEW

File: virtualenv.py [hunk 1]
```diff
@@ -1593,10 +1607,12 @@ def resolve_interpreter(exe):
...
-            if os.path.exists(os.path.join(path, exe)):
-                exe = os.path.join(path, exe)
+            check_exe = resolve_exe(os.path.join(path, exe))
+            if check_exe:
+                exe = check_exe
                 break
-    if not os.path.exists(exe):
+    exe = resolve_exe(exe)
+    if not exe:
         logger.fatal('The executable %s (from --python=%s) does not exist' % (exe, exe))
```
Comment (pfmoore) [inline virtualenv.py]: By overwriting ```exe``` here you've made the error message misleading (as it no longer states the same argument as the user supplied).

File: virtualenv.py [hunk 2]
```diff
@@ -1580,6 +1580,20 @@ def fix_lib64(lib_dir, symlink=True):
...
 
+def resolve_exe(exe):
+    if os.path.exists(exe):
+        return exe
+
+    if os.name != 'nt':
+        return None
+
+    # check PATHEXT
+    exts = os.environ['PATHEXT'].split(os.pathsep)
```
Comment (pfmoore) [inline virtualenv.py]: Do we really want to use ```PATHEXT``` here? If what you've found isn't ```python.exe``` it's unlikely to actually be a valid Python installation, after all...

**Comment by BrownTruck** [comment]
Hello!

As part of an effort to ease the contribution process and adopt a more standard workflow pip has switched to doing development on the ``master`` branch. However, this Pull Request was made against the ``develop`` branch so it will need to be resubmitted against ``master``. Unfortunately, this pull request does not cleanly merge against the current ``master`` branch.

If you do nothing, this Pull Request will be automatically closed by @BrownTruck since it cannot be merged.

If this pull request is still valid, please rebase it against ``master`` (or merge ``master`` into it) and resubmit it against the ``master`` branch, closing and referencing the original Pull Request.

If you choose to rebase/merge and resubmit this Pull Request, here is an example message that you can copy and paste:

```
This fixes #640 for me.

It works now with virtualenv --python=C:\Python27\python and virtualenv --python=python on Windows.

---

*This was migrated from pypa/virtualenv#641 to reparent it to the ``master`` branch. Please see original pull request for any previous discussion.*
```

**Comment by BrownTruck** [comment]
This Pull Request was closed because it cannot be automatically reparented to the ``master`` branch and it appears to have bit rotted.

Please feel free to re-open it or re-submit it if it is still valid and you have rebased it onto ``master`` or merged ``master`` into it.

**Pull request closed without merging**